core/iwasm/libraries/debug-engine/handler.c

```diff
@@ -11,18 +11,30 @@
 #include "wasm_runtime.h"
 
 #define MAX_PACKET_SIZE (0x20000)
-static char tmpbuf[MAX_PACKET_SIZE];
+static char *tmpbuf;
 static korp_mutex tmpbuf_lock;
 
 int
 wasm_debug_handler_init()
 {
-    return os_mutex_init(&tmpbuf_lock);
+    int ret;
+    tmpbuf = wasm_runtime_malloc(MAX_PACKET_SIZE);
```

Comment:
On some constraint devices, there is a great possibility of malloc failure for such a large space, had better add `LOG_ERROR("Allocate debug package buffer failed");`, otherwise the developer may not know what happened

Reply:
done.
btw, do you (or anyone) remember why this huge buffer is allocated here?
is it somehow required by the protocol?